### ISSUE TITLE
Support CommonJS

### ISSUE THREAD

**Issue by bendrucker** [firebase/angularfire#566]
Some upstanding citizen who may or may not be me shipped CommonJS support for Angular in 1.3.14 and 1.4 beta. So that means you can `require('angular')`, but also this:

```js
require('angular').module('myApp', [
  require('angular-messages')
]);
```

That's because `require('angular-messages') === 'ngMessages'`. Requiring the code registers the module, but the actual return value is the string name of the module.

Here's how to do this without actually using browserify to build AngularFire:

1. Change `main` in `package.json` to `index.js`
2. Create `index.js` at the project root
3. Populate `index.js` with the following:

```js
require('./dist/angularfire.js');
module.exports = 'firebase';
```

This is exactly how Angular does it for all its various `angular-*` repos. 

Full discussion where this was discussed and my way was blessed in https://github.com/angular/angular.js/pull/10732

**Comment by jamestalmage** [comment]
lol

![screenshot 2015-02-26 13 16 17](https://cloud.githubusercontent.com/assets/4082216/6398477/e7b5a76c-bdb9-11e4-8073-8b39de10c9a9.png)

**Comment by jwngr** [comment]
Man on a mission.

**Comment by bendrucker** [comment]
I had to write bash for this. Shit got real: https://github.com/angular/bower-angular-i18n/commit/8e4b78db276b5cfdbcf44c1f26ae6f5cd6a180f4

Also:

![holy mother of files](http://f.cl.ly/items/3U0y3i2q0814362U2h2O/Screen%20Shot%202015-02-26%20at%204.35.38%20PM.png)

**Comment by bendrucker** [comment]
Merged in #567

**Comment by jwngr** [comment]
Haha, thanks @bendrucker; nice addition! I added it as a feature for [our running changelog for 1.0.0](https://github.com/firebase/angularfire/pull/561).
